pkg/apis/pipeline/v1/matrix_types.go

```diff
@@ -360,3 +332,21 @@ func (m *Matrix) validateParameterInOneOfMatrixOrParams(params []Param) (errs *a
 	}
 	return errs
 }
+
+// validateNoWholeArrayResults() is used to ensure a matrix parameter does not contain result references
+// to entire arrays. This is temporary until whole array replacements for matrix paraemeters are supported.
+// See issue #6056 for more details
+func (m *Matrix) validateNoWholeArrayResults() (errs *apis.FieldError) {
+	if m.HasParams() {
+		for i, param := range m.Params {
+			val := param.Value.StringVal
+			expressions, ok := GetVarSubstitutionExpressionsForParam(param)
+			if ok {
+				if LooksLikeContainsResultRefs(expressions) && strings.Contains(val, "[*]") {
```

Comment:
This still seems like it could be error prone. would this function work here? https://github.com/tektoncd/pipeline/blob/7d6cd810e2ea4348415be9b094d1db9deb625e3e/pkg/apis/pipeline/v1beta1/resultref.go#L194-L205